runtime/modules/ui.lua

```diff
@@ -0,0 +1,81 @@
+local vscode = require("vscode-neovim")
+
+-- vim.ui.select({items}, {opts}, {on_choice})                  *vim.ui.select()*
+--     Prompts the user to pick from a list of items, allowing arbitrary
+--     (potentially asynchronous) work until `on_choice`.
+
+--     Example: >lua
+--         vim.ui.select({ 'tabs', 'spaces' }, {
+--             prompt = 'Select tabs or spaces:',
+--             format_item = function(item)
+--                 return "I'd like to choose " .. item
+--             end,
+--         }, function(choice)
+--             if choice == 'spaces' then
+--                 vim.o.expandtab = true
+--             else
+--                 vim.o.expandtab = false
+--             end
+--         end)
+-- <
+--     Parameters: ~
+--       • {items}      (table) Arbitrary items
+--       • {opts}       (table) Additional options
+--                      • prompt (string|nil) Text of the prompt. Defaults to
+--                        `Select one of:`
+--                      • format_item (function item -> text) Function to format
+--                        an individual item from `items`. Defaults to
+--                        `tostring`.
+--                      • kind (string|nil) Arbitrary hint string indicating the
+--                        item shape. Plugins reimplementing `vim.ui.select` may
+--                        wish to use this to infer the structure or semantics of
+--                        `items`, or the context in which select() was called.
+--                      • match_on_detail (boolean|nil) optional flag to use the detail when filtering the items. items must have a detail field.
```

Comment:
It is unnecessary to fully match the options of VSCode's QuickPick, as this is not very useful and users will not even know about this option unless it is explicitly mentioned in the documentation. I suggest keeping the signature of the original vim.ui function consistent.

Reply:
I chose to include the match_on_detail as it is a useful vscode feature available. I would prefer to document it and make it available rather than limiting it to what the vim.ui.select spec currently supports. In my opinion there is already precedent, as the vim.ui.select function takes a kind and format function so that you can pass your own custom items. Passing vscode specific options to improve the vscode experience seems reasonable to me.

Reply:
It is a burden for both users and maintainers.

Reply:
Take a look at the update and let me know your thoughts - as a middle ground, I removed the extra options, but I allow passing through 'detail' when users provide tables instead of just strings. This keeps the API signature of vim.ui.select unchanged, but, still enables users to take advantage of the two-line label/detail display with the filtering available in the vscode QuickPick.